src/python/Publisher/PublisherMaster.py

```diff
@@ -216,49 +216,52 @@ def active_tasks(self, db):
         """
 
         self.logger.debug("Retrieving publications from oracleDB")
+        filesToPublish = []
         uri = self.REST_filetransfers
-        fileDoc = {}
-        fileDoc['asoworker'] = self.config.asoworker
-        fileDoc['subresource'] = 'acquirePublication'
-        data = encodeRequest(fileDoc)
-        results = ''
-        try:
-            results = db.post(uri=uri, data=data)
-        except Exception as ex:
-            self.logger.error("Failed to acquire publications from %s: %s", uri, ex)
-            return []
-
-        self.logger.debug("Retrieving max.100000 acquired publications from oracleDB")
-        fileDoc = dict()
-        fileDoc['asoworker'] = self.config.asoworker
-        fileDoc['subresource'] = 'acquiredPublication'
-        fileDoc['grouping'] = 0
-        fileDoc['limit'] = 100000
-        result = []
-        uri = self.REST_filetransfers
-        data = encodeRequest(fileDoc)
-
-        try:
-            results = db.get(uri=uri, data=data)
-            result.extend(oracleOutputMapping(results))
-        except Exception as ex:
-            self.logger.error("Failed to acquire publications from %s: %s", uri, ex)
-            return []
-
-        self.logger.debug("publen: %s", len(result))
+        asoworkers = self.config.asoworker
+        # asoworkers can be a string or a list of strings
+        # but if it is a string, do not turn it into a list of chars !
+        asoworkers = [asoworkers] if isinstance(asoworkers, basestring) else asoworkers
+        for asoworker in asoworkers:
+            self.logger.info("Processing publication requests for asoworker: %s", asoworker)
+            fileDoc = {}
+            fileDoc['asoworker'] = asoworker
+            fileDoc['subresource'] = 'acquirePublication'
+            data = encodeRequest(fileDoc)
+            try:
+                result = db.post(uri=uri, data=data)
+            except Exception as ex:
+                self.logger.error("Failed to acquire publications from %s: %s", uri, ex)
+                return []
+
+            self.logger.debug("Retrieving max.100000 acquired publications from oracleDB")
+            fileDoc = dict()
+            fileDoc['asoworker'] = asoworker
+            fileDoc['subresource'] = 'acquiredPublication'
+            fileDoc['grouping'] = 0
+            fileDoc['limit'] = 100000
+            uri = self.REST_filetransfers
```

Comment:
why this line is needed if that was already done in line 220?

Reply:
indeed it is not needed. It was simply for consistency, so that the piece of code that executes a given action is complete and can e.g. safely be moved around. I think earlier implementation by Diego has the uri path extplicitely writtin in the db.post/get call, which I did not like so I created the `uri` variable. All in all I think it is good that righe before making a call to the db one define explicitely which API will be used.
Also `fileDoc = {}` and `fileDoc = dict()` could be made uniform !

Reply:
sorry, I clicked on start-review instead of submit-comment. was not trying to confuse you !